TracerLib/TracerImpl/Tracer.cs

```diff
@@ -44,49 +38,53 @@ private Tracer()
 
         public void StartTrace()
         {
-            StackTrace st = new StackTrace();
-            var invokationFrame = (st.GetFrames())[1];
             var info = new TracedMethodInfo
             {
-                ClassName = invokationFrame.GetMethod().DeclaringType.Name,
-                MethodName = invokationFrame.GetMethod().Name,
-                ParamCountInMethod = invokationFrame
-                    .GetMethod()
-                    .GetParameters()
-                    .Count(),
-                Time = Stopwatch.StartNew()
+               Time = Stopwatch.StartNew()
             };
             _current = _current.AddChild(info);
             _methodsStack.Push(_current);
         }
 
         public void StopTrace()
         {
-            _methodsStack.Pop().Data.Time.Stop(); ;
-            _current = _methodsStack.Count==0 ? _tree : _methodsStack.Peek();
+            var st = new StackTrace();
+            var invokationFrame = (st.GetFrames())[1];
+            var info = _methodsStack.Pop().Data;
+            info.ClassName = invokationFrame.GetMethod().DeclaringType.Name;
```

Comment:
GetMethod and DeclaringType could throw null reference exception. Check for null.